### PR TITLE
Add projector.next action + tests.

### PR DESCRIPTION
**Pull request by reiterl**

### CODE REVIEW

File: tests/system/action/projector/test_next.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class ProjectorNext(BaseActionTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.set_models(
+            {
+                "meeting/1": {},
+                "projector/2": {"meeting_id": 1},
+                "projector/3": {
+                    "current_projection_ids": [1, 2],
```
Comment (FinnStutzenstein) [inline tests/system/action/projector/test_next.py:12]: Please add a third stable element and test, that it stays projected

File: tests/system/action/projector/test_next.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    "preview_projection_ids": [3, 4],
+                    "history_projection_ids": [],
```
Comment (FinnStutzenstein) [inline tests/system/action/projector/test_next.py:14]: Add an element here, so you can check the weight handling.

File: tests/system/action/projector/test_next.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        assert projection.get("preview_projection_ids") is None
+        assert projection.get("history_projection_ids") is None
+
+    def test_next_complex(self) -> None:
+        response = self.request("projector.next", {"id": 3})
+        self.assert_status_code(response, 200)
+        projection = self.get_model("projector/3")
+        assert projection.get("current_projection_ids") == [3]
+        assert projection.get("preview_projection_ids") == [4]
+        assert projection.get("history_projection_ids") == [1, 2]
```
Comment (FinnStutzenstein) [inline tests/system/action/projector/test_next.py:45]: also test all weight (maybe not in this test): Projections added to the history must get `max+1`

File: tests/system/action/projector/test_next.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                "projection/3": {"preview_projector_id": 3, "meeting_id": 1},
+                "projection/4": {"preview_projector_id": 3, "meeting_id": 1},
+                "projection/5": {"preview_projector_id": 4, "meeting_id": 1},
+            }
+        )
+
+    def test_next_nothing(self) -> None:
+        response = self.request("projector.next", {"id": 2})
+        self.assert_status_code(response, 200)
+        projection = self.get_model("projector/2")
```
Comment (FinnStutzenstein) [inline tests/system/action/projector/test_next.py:34]: `projection` -> `projector` (In all tests)

File: openslides_backend/action/actions/projector/next.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                result = self.datastore.get_many([gmr])
+                current_projections = list(
+                    result.get(Collection("projection"), {}).values()
+                )
+            new_current_projection_ids = [
+                projection["id"]
+                for projection in current_projections
+                if self.is_stable(projection)
+            ]
+            new_history_projection_ids = projector.get("history_projection_ids", []) + [
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/projector/next.py:49]: I think the `weight` handling is missing.

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/projector/next.py (hunk 5 above)
Comment (FinnStutzenstein) [reply openslides_backend/action/actions/projector/next.py:49]: (See https://github.com/OpenSlides/OpenSlides/wiki/Projection#preview-and-history)

Comment (reiterl) [reply openslides_backend/action/actions/projector/next.py:49]: Add the weight handling and update the test.

File: tests/system/action/projector/test_next.py (hunk 4 above)
Comment (reiterl) [reply tests/system/action/projector/test_next.py:34]: Yes, fix that typos.

File: tests/system/action/projector/test_next.py (hunk 3 above)
Comment (reiterl) [reply tests/system/action/projector/test_next.py:45]: Changed the structure of the test and add a weight test.

File: tests/system/action/projector/test_next.py (hunk 1 above)
Comment (reiterl) [reply tests/system/action/projector/test_next.py:12]: Set the second element to stable and test this function.

File: tests/system/action/projector/test_next.py (hunk 2 above)
Comment (reiterl) [reply tests/system/action/projector/test_next.py:14]: Add weight to the projection/1 to check the weight.

Comment (FinnStutzenstein) [reply tests/system/action/projector/test_next.py:14]: Also add a new projection to the history with some weight and check, that the order after calling `next` is right.

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]
The weight part seems to be ok, some remarks are still left.

File: openslides_backend/action/actions/projector/next.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                result = self.datastore.get_many([gmr])
+                current_projections = list(
+                    result.get(Collection("projection"), {}).values()
+                )
+            new_current_projection_ids = [
+                projection["id"]
+                for projection in current_projections
+                if self.is_stable(projection)
+            ]
+            transfer_projections = [
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/projector/next.py:52]: Renamed.

File: openslides_backend/action/actions/projector/next.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                projection["id"]
+                for projection in current_projections
+                if not self.is_stable(projection)
+            ]
+            new_history_projection_ids = (
+                projector.get("history_projection_ids", []) + transfer_projections
+            )
+            self.set_weight_to_projection(transfer_projections, projector["meeting_id"])
+            transfer_id = self.get_min_preview_projection(projector)
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/projector/next.py:61]: Renamed

File: tests/system/action/projector/test_next.py (hunk 2 above)
Comment (reiterl) [reply tests/system/action/projector/test_next.py:14]: Add a new projection to history of projector 3.  Add weight to the projections.

File: openslides_backend/action/actions/projector/next.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+            ]
+            instance["current_projection_ids"] = new_current_projection_ids
+            instance["preview_projection_ids"] = new_preview_projection_ids
+            instance["history_projection_ids"] = new_history_projection_ids
+            yield instance
+
+    def is_stable(self, value: Dict[str, Any]) -> bool:
+        return bool(value.get("stable"))
+
+    def get_max_projection_weight(self, meeting_id: int) -> int:
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/projector/next.py:74]: Use `max+1` of all history projections of the projector.

File: tests/system/action/projector/test_next.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+class ProjectorNext(BaseActionTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.set_models(
+            {
+                "meeting/1": {},
+                "projector/2": {"meeting_id": 1},
+                "projector/3": {
+                    "current_projection_ids": [1, 2],
+                    "preview_projection_ids": [3, 4],
```
[the remark this answers is not in the document]
Comment (reiterl) [reply tests/system/action/projector/test_next.py:13]: Done.

**Review by FinnStutzenstein — APPROVED** [review]

**Comment by FinnStutzenstein** [comment]
@jsangmeister or @r-peschke Can someone of you fix the CI, so this PR can be merged?

**Pull request merged**